apps/native/app/src/screens/vaccinations/vaccinations.tsx

```diff
@@ -0,0 +1,137 @@
+`
+
+const Vaccinations = styled.View`
+  margin-top: ${({ theme }) => theme.spacing[3]}px;
+`
+
+const Tabs = styled.View`
+  margin-top: ${({ theme }) => theme.spacing[3]}px;
+`
+
+const { getNavigationOptions, useNavigationOptions } =
+  createNavigationOptionHooks((theme, intl) => ({
+    topBar: {
+      title: {
+        text: intl.formatMessage({ id: 'health.vaccinations.screenTitle' }),
+      },
+    },
+    bottomTabs: {
+      visible: false,
+      drawBehind: true,
+    },
+  }))
+
+export const VaccinationsScreen: NavigationFunctionComponent = ({
+  componentId,
+}) => {
+  useNavigationOptions(componentId)
+  const intl = useIntl()
+  const [refetching, setRefetching] = useState(false)
+  const [selectedTab, setSelectedTab] = useState(0)
+
+  const vaccinationsRes = useGetVaccinationsQuery()
+
+  const vaccinations = vaccinationsRes.data?.healthDirectorateVaccinations
+    .vaccinations
+    ? selectedTab === 0
+      ? vaccinationsRes.data?.healthDirectorateVaccinations.vaccinations.filter(
+          (v) => v.isFeatured,
+        )
+      : vaccinationsRes.data?.healthDirectorateVaccinations.vaccinations.filter(
+          (v) => !v.isFeatured,
+        )
+    : []
+
+  useConnectivityIndicator({
+    componentId,
+    refetching,
+    queryResult: vaccinationsRes,
+  })
+
+  const onRefresh = useCallback(async () => {
+    setRefetching(true)
+
+    try {
+      await vaccinationsRes.refetch()
+    } catch (e) {
+      // noop
+    } finally {
+      setRefetching(false)
+    }
+  }, [vaccinationsRes])
+
+  return (
+    <View style={{ flex: 1 }}>
+      <ScrollView
+        refreshControl={
+          <RefreshControl refreshing={refetching} onRefresh={onRefresh} />
+        }
+        style={{ flex: 1 }}
+      >
+        <Host>
+          <Heading>
+            <FormattedMessage
+              id="health.vaccinations.title"
+              defaultMessage="Bólusetningar"
+            />
+          </Heading>
+          <Typography>
+            <FormattedMessage
+              id="health.vaccinations.description"
+              defaultMessage="Hér getur þú séð lista yfir bóluefni sem þú hefur fengið, stöðu bólusetningar og aðrar upplýsingar."
+            />
+          </Typography>
+          <Tabs>
+            <TabButtons
+              buttons={[
+                {
+                  title: intl.formatMessage({
+                    id: 'health.vaccinations.generalVaccinations',
+                  }),
+                },
+                {
+                  title: intl.formatMessage({
+                    id: 'health.vaccinations.otherVaccinations',
+                  }),
+                },
+              ]}
+              selectedTab={selectedTab}
+              setSelectedTab={setSelectedTab}
+            />
+          </Tabs>
+          <Vaccinations>
+            {vaccinationsRes.loading && !vaccinationsRes.data
+              ? Array.from({ length: 5 }).map((_, index) => (
```

Comment:
nit: It would be nice since the Array from is in a lot of places in this project to have a util function that does this something like

```
const createArray = <T>(size: number, mapper: (index: number) => T): T[] =>
  Array.from({ length: size }, (_, index) => mapper(index))

createArray(5, (index) => (
  <GeneralCardSkeleton height={90} key={index} />
))
```